python/lbann/torch/compiler.py

```diff
@@ -0,0 +1,199 @@
+################################################################################
+# Copyright (c) 2014-2023, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+# Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+# the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+#
+# LLNL-CODE-697807.
+# All rights reserved.
+#
+# This file is part of LBANN: Livermore Big Artificial Neural Network
+# Toolkit. For details, see http://software.llnl.gov/LBANN or
+# https://github.com/LLNL/LBANN.
+#
+# Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+# may not use this file except in compliance with the License.  You may
+# obtain a copy of the License at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the license.
+#
+################################################################################
+"""
+Provides an API for LBANN graph compilation with PyTorch 2.x.
+"""
+
+# Version check
+try:
+    import torch
+    if int(torch.__version__.split('.')[0]) < 2:
+        raise ImportError(
+            'PyTorch 2.0 or newer is required for the LBANN torch frontend. '
+            f'Found version {torch.__version__}')
+except (ModuleNotFoundError, ImportError):
+    raise ImportError(
+        'PyTorch (version 2.0 or newer) is required for the LBANN torch '
+        'frontend.')
```

Comment:
Does this actually ensure we have Dynamo? I thought you had to have a nightly build or build from source, even with 2.0, to get it.

Reply:
It exists when pip installing it. Not sure otherwise.